src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -148,6 +158,32 @@ public void visitJavadocToken(DetailNode ast) {
     }
 
     /**
+     * Finds if inheritDoc is placed properly in java doc.
+     * @param firstSentence first sentence of java doc.
+     * @return true if inheritDoc is valid or fasle.
+     */
+    private static boolean isValidInheritDocAfterRemovingAsteric(String firstSentence) {
+        final int begin = firstSentence.indexOf(INHERIT_DOC_LITERAL);
+        final int end = firstSentence.indexOf("<EOF>");
+        boolean result = false;
+        if (begin == -1) {
+            result = false;
+        }
+        final StringBuilder innerContent = new StringBuilder();
```

Comment:
looks like you need to finish method execution at this point or all further statements should be in ELSE block.